Server/DeviceManagement/DeviceManagement.Domain/UserAggregate/IUserRepository.cs

```diff
@@ -0,0 +1,12 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace ModularHouse.Server.DeviceManagement.Domain.UserAggregate;
+
+public interface IUserRepository
+{
+    public Task CreateAsync(User user, CancellationToken cancellationToken);
+    public Task DeleteAsync(Guid id, CancellationToken cancellationToken);
```

Comment:
As we discussed, repository should not contain `if is null` conditions inside; In that case it's better to delete entity not by `Id`, but by passing directly the whole entity
Update `DeleteAsync(Guid id, ...)` to `DeleteAsync(User user, ...)`